apps/web/src/pages/settings/user-profile-page/useUserProfileSetPassword.ts

```diff
@@ -0,0 +1,58 @@
+import { showNotification } from '@mantine/notifications';
+import { IResponseError, PasswordResetFlowEnum } from '@novu/shared';
+import { api, useAuthContext } from '@novu/shared-web';
+import { useMutation } from '@tanstack/react-query';
+import { useCountdownTimer } from '../../../hooks';
+
+const COUNTDOWN_TIMER_START_SECONDS = 60;
+
+export const useUserProfileSetPassword = () => {
+  const { currentUser } = useAuthContext();
+  const {
+    timeRemaining: countdownTimerSeconds,
+    resetTimer,
+    stopTimer,
+  } = useCountdownTimer({
+    // start at 0 in the event the user opens the page directly to the set password flow via URL
+    startTimeSeconds: 0,
+  });
+
+  const { mutateAsync } = useMutation<
+    { success: boolean },
+    IResponseError,
+    {
+      email: string;
+    }
+  >((data) => api.post(`/v1/auth/reset/request`, data, { src: PasswordResetFlowEnum.USER_PROFILE }), {
+    onError: (err) => {
+      showNotification({
+        color: 'red',
+        message: `There was an error sending your email. Please refresh and try again. Error: ${err.message}`,
+      });
+    },
+  });
```

Comment:
<!-- TODO's are small, trivial, but necessary changes. Distinguishing todo comments from issues: or suggestions: helps direct the reader's attention to comments requiring more involvement.  -->
**todo:** You can use `errorMessage` and `successMessage` notification helpers from our design-system‏

Reply:
![image](https://github.com/novuhq/novu/assets/29251776/1436f51e-7c2b-4365-977f-79e2a466c524)
The long error message also looks bad on the UI. Not sure if it using the one from the design-system will make it better.

Reply:
Thanks for letting me know about these utils! I hadn't seen them, and the code I was looking at used `showNotification` directly. Will update